scalameter-core/src/main/java/org/scalameter/execution/invocation/instrumentation/MethodInvocationCounter.java

```diff
@@ -0,0 +1,65 @@
+package org.scalameter.execution.invocation.instrumentation;
+
+import java.util.concurrent.atomic.AtomicLongArray;
+
+
+/**
+ * INTERNAL API
+ *
+ * Stores method invocations in a AtomicLongArray.
+ */
+public class MethodInvocationCounter {
+    private static volatile boolean _initialized = false;
+    private static volatile boolean _started = false;
+    private static AtomicLongArray _counts;
+
+    /**
+     * Initializes count array.
+     *
+     * @param numMethods length of an array
+     */
+    public static synchronized void setup(int numMethods) {
+        _counts = new AtomicLongArray(numMethods);
+        _initialized = true;
+    }
+
+    /**
+     * Starts measuring of method invocations.
+     */
+    public static synchronized void start() {
+        if (_initialized && !_started) {
+            _started = true;
+        }
```

Comment:
Well, since this method is intended only to be used by invocation counting measurers I thought that it isn't necessary to add any alternative flow.
So maybe the better idea would be to make all this methods (except `methodCalled`) package-restricted and write Scala wrapper inside that package that would call this methods (and it would have appropriate, like `private[scalamater]` access level)?